rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1115,14 +1115,16 @@ public NodeSet getExactlyNodes(int nb, SelectionScript selectionScript) {
      * @return RMInitialState containing nodes and nodeSources of the RMCore.
      */
     public RMInitialState getRMInitialState() {
+        Collection<RMNode> nodes = this.allNodes.values();
+        ArrayList<RMNodeEvent> nodesList = new ArrayList<>(nodes.size());
 
-        ArrayList<RMNodeEvent> nodesList = new ArrayList<>();
-        for (RMNode rmnode : this.allNodes.values()) {
+        for (RMNode rmnode : nodes) {
             nodesList.add(rmnode.createNodeEvent());
         }
 
-        ArrayList<RMNodeSourceEvent> nodeSourcesList = new ArrayList<>();
-        for (NodeSource s : this.nodeSources.values()) {
+        Collection<NodeSource> nodeSources = this.nodeSources.values();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "nodeSources" which hides the field declared at line 166. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)